client/jetpack-dashboard/controller.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal Dependencies
+ */
+import JetpackDashboardSecurity from './security';
+import JetpackDashboardSidebar from './sidebar';
+import { preload } from 'sections-helper';
+
+export function preloadJetpackDashboard( context, next ) {
+	preload( 'jetpack-dashboard' );
```

Comment:
This preloading code shouldn't be necessary. The server should send all the section's chunks in the _index.html_ document. You updated the server code that loads the right sections and handles their routes, so it should work out of the box.